apis/taskcluster.py

```diff
@@ -52,3 +64,132 @@ def submit_to_try(self, library):
+
+    # =================================================================
+    # =================================================================
+
+    @logEntryExit
+    def get_job_details(self, revision):
+        push_list_url = self.url_treeherder + "/api/" + self.project + "push/?revision=%s" % revision
+        self.logger.log("Requesting revision %s from %s" % (revision, push_list_url), level=LogLevel.Info)
+
+        r = requests.get(push_list_url, headers=self.HEADERS)
+        push_list = r.json()
+        try:
+            push_id = push_list['results']['id']
+        except Exception as e:
+            raise Exception("Could not find the expected ['results']['id'] from %s" % r.text) from e
+
+        job_list = []
+        property_names = []
+        job_details_url = self.url_treeherder + "/api/jobs/?push_id=%s" % push_id
+        try:
+            while job_details_url:
+                self.logger.log("Requesting push id %s from %s" % (push_id, job_details_url), level=LogLevel.Info)
+                r = requests.get(job_details_url, headers=self.HEADERS)
+                j = r.json()
+
+                job_list.extend(j['results'])
+                if not property_names:
+                    property_names = j['job_property_names']
+                else:
+                    for i in range(len(property_names)):
+                        if len(property_names) != len(j['job_property_names']):
+                            raise Exception("The first j['job_property_names'] was %i elements long, but a subsequant one was %i for url %s" % (len(property_names), len(j['job_property_names']), job_details_url))
+                        elif property_names[i] != j['job_property_names'][i]:
+                            raise Exception("Property name %s (index %i) doesn't match %s" % (property_names[i], i, j['job_property_names'][i]))
+
+                job_details_url = j['next'] if 'next' in j else None
+        except Exception as e:
+            raise Exception("Could not obtain all the job results for push id %s" % push_id) from e
+
+        new_job_list = TaskclusterProvider._transform_job_list(property_names, job_list)
+
+        return new_job_list
+
+    # =================================================================
+    # =================================================================
+
+    @logEntryExitNoArgs
+    def retrigger_jobs(self, job_list, retrigger_list):
+        # Go through the jobs and find the decision task
+        decision_task = None
+        for j in job_list:
+            if "Gecko Decision Task" == j.job_type_name:
+                decision_task = j
+                break
+        assert decision_task is not None
+
+        # Download its actions.json
+        artifact_url = self.url_taskcluster + "api/queue/v1/task/%s/runs/0/artifacts/public/actions.json" % (decision_task.task_id)
+        r = requests.get(artifact_url, headers=self.HEADERS)
+        actions = r.json()
```

Comment:
It looks like we might not be getting json back in the response here. Showing up in the testRetrigger test on line 112.

```======================================================================
ERROR: testRetrigger (tests.treeherder_api.TestTaskclusterProvider)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jewilde/Source/updatebot/tests/treeherder_api.py", line 112, in testRetrigger
    decision_task = self.taskclusterProvider.retrigger_jobs(job_list, to_retrigger)
  File "/Users/jewilde/Source/updatebot/components/logging.py", line 31, in func_wrapper
    ret = func(*args, **kwargs)
  File "/Users/jewilde/Source/updatebot/apis/taskcluster.py", line 161, in retrigger_jobs
    actions = r.json()
  File "/Users/jewilde/Library/Caches/pypoetry/virtualenvs/updatebot-6T8aijfh-py3.8/lib/python3.8/site-packages/requests/models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)```

Reply:
Yeah this is the behavior we get when we get a 404 or other HTTP error.  I'm going to add a commit to handle this type of error all over the place....

Reply:
I didn't add it in a separate commit, instead I added it in-line